### PR TITLE
feat: sync progress stepper with changes from skin

### PR DESCRIPTION
**Pull request by agliga**
## Description
* Synced progress stepper with skin changes
* Added migrator for `ebay-stepper` to point to `progresss-stepper`
* Cleaned up progress stepper tests to use snapshots
* Fixed a couple of minor bugs with progress stepper (like reseting current state when default state is set)

## References
https://github.com/eBay/ebayui-core/issues/1705

## Screenshots
Screen Shot 2022-06-29 at 9 17 31 AM
Screen Shot 2022-06-29 at 9 17 38 AM
Screen Shot 2022-06-29 at 9 17 15 AM

### CODE REVIEW

**Comment by agliga** [comment]
> Looks good! The only input I have is that it might be worth cleaning up [this section](https://github.com/eBay/ebayui-core/pull/1706/files#diff-df78e73911ec8794a0ed91776939451334d1aa7329bb4928819d0fb2517cefe7R20-R28) since `findIndex` is called unnecessarily in the case of a "complete" or "upcoming" state, but the array will always be so small that this is negligible for performance.



> Looks good! The only input I have is that it might be worth cleaning up [this section](https://github.com/eBay/ebayui-core/pull/1706/files#diff-df78e73911ec8794a0ed91776939451334d1aa7329bb4928819d0fb2517cefe7R20-R28) since `findIndex` is called unnecessarily in the case of a "complete" or "upcoming" state, but the array will always be so small that this is negligible for performance.

Good catch. Updated PR with a change to optimize it.

**Review by LuLaValva — APPROVED** [review]

**Comment by ArtBlue** [comment]
I caught what I think is a bug. It possibly came in with the Skin PR. According to the Skin issue specs, the font weight for all other step labels is supposed to be "normal" except for current step. It looks like they're all bold now. Here's my local:  
image

**Review by ArtBlue — COMMENTED** [review]
I caught what I think is a bug. It possibly came in with the Skin PR. According to the Skin issue specs, the font weight for all other step labels is supposed to be "normal" except for current step. It looks like they're all bold now. Here's my local: 
image

File: src/components/ebay-icon/icons/ebay-stepper-current-icon/marko-tag.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "attribute-groups": ["html-attributes"],
+  "@*": {
+    "targetProperty": null,
+    "type": "expression"
+  },
+  "@html-attributes": "expression",
+  "@a11y-text": "string",
+  "@a11y-variant": "string",
+  "@no-skin-classes": "boolean",
+  "@role": "never",
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/components/ebay-icon/icons/ebay-stepper-current-icon/marko-tag.json:12]: Yes! We use it when we don't want people overriding something. Like in this case we have blocked aria role which should never change.

File: src/components/ebay-progress-stepper/index.marko [hunk 2]
```diff
@@ -58,21 +73,24 @@ $ var current = (input.step || []).findIndex(item => item.current);
                     <else-if(item.type === "information")>
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/components/ebay-progress-stepper/index.marko:73]: The problem is that a lot of these have various logic based around count and or input variables. Its hard to replace it with a case statement. 
That said we can probably clean it up.

**Comment by ArtBlue** [comment]
> > I caught what I think is a bug. It possibly came in with the Skin PR. According to the Skin issue specs, the font weight for all other step labels is supposed to be "normal" except for current step. It looks like they're all bold now. Here's my local:
> > image
> 
> Did you link 14.1.0 skin? If you look at my screenshots the font size and such is present. Let me know if you did and if its still not showing up.

I actually checked out `14.1.0`, cleaned, installed, and started the Skin site too, and I see the same thing there. Very strange. 
image

If Skin and UI Core look proper on your end, I'll assume something odd is happening on my end.

**Comment by ArtBlue** [comment]
> Did you pull the latest code from github? That screenshot from skin site looks like it doenst have my changes.

(egg on face) 🤦 I had a local checked out and forgot to pull. Sorry. Looks good.

**Review by ArtBlue — APPROVED** [review]
Looks good. I'll leave the `else-if` cleanup up to you.

**Pull request merged**